lib/printer.js

```diff
@@ -556,11 +561,18 @@ function genericPrintNoParens(path, options, print) {
                 // both (all) of the holes.
                 parts.push(",");
             } else {
-                if (i > 0)
-                    parts.push(" ");
-                parts.push(print(elemPath));
-                if (i < len - 1)
+                var lines = print(elemPath);
```

Comment:
Oh, yeah, reusing the printed elements (`Lines` objects) from above should be feasible.